### ISSUE TITLE
Feature request: Upgrade to GPT-3

### ISSUE THREAD

**Issue by Belzedar94** [Vayku/Hanbot#1]
Hello buddy,

First of all I want to tell you, this is the best script I have seen in my whole life, and I've been scripting for so long I can't remember. It is really really fun when people start engaging in conversations with the bot and how they react.

This issue is not really an issue but a feature request. Cleverbot has a really old AI in use, so I thought it would be a huge upgrade if the script can be upgraded into a GPT-3 chatbot. I'm not a programmer so I don't know how difficult it is, but if it is easy to do (I think there is available API and utilities to make it simple), then it would be a huge upgrade for your script.

Thank you and keep it up!

**Comment by Belzedar94** [comment]
A nice example:

https://github.com/plotly/dash-sample-apps/tree/main/apps/dash-gpt3-chatbot

**Comment by Vayku** [comment]
Hi Belzedar,

Awesome! I am glad to hear the script is entertaining, I certainly loved the random engagement people have with it. The request to implement an alternative based on the github example you send me is certainly do-able, I have since applied for OpenAI.

I have seen other GPT-3 alternatives that also caught my interest, all of them way more sophisticated than Clever AI. 
Great request and i'll certainly look more into the topic.

**Comment by Belzedar94** [comment]
Happy to hear that! By the way, I already have an openAI API key, so I can share it with you :)

You can hit me up in discord: Belzedar#8832

**Comment by Belzedar94** [comment]
Hey buddy! Did you had the time to check it? I was doing some tests with my OpenAI access and it was hilarious!

Also, I'd like to request another feature, would be nice to have an option to answer allies through the team chat instead of /all.

Thank you and keep it up :)

**Comment by Vayku** [comment]
Hey Belzedar!

Apologies for the very delayed response, I haven't really used HB in a while due to the limited playtime it upholds upon injecting. 

I did get my hands on the OpenAI API key, though after reading the restrictions it applies, specifically towards Chatbots it's unlikely it will work for an extended period of time. To my surprise a lot of people actually use it 😄 

Adding the last request however would only take a couple of minutes. I'll see if I can reach out to Han to get a new developer key.

**Comment by Belzedar94** [comment]
Hey buddy! The Chatbot is not working anymore sadly :(

I hope you have the motivation to fix it soon.

Thanks :D

**Comment by Belzedar94** [comment]
Also, attaching below an implementation of a free gpt-like bot, in case you find it useful if at some point you'd like to improve the chatbot. I genuinely think it's the funniest script I have ever tried!!

Thank you! :)

**Comment by Joey0x646576** [comment]
Hey there! 

It seems like Cleverbot changed something in their approach, which broke some back-end related calls. I applied a temporary workaround, which does not need an update to the script.

I might just waste a night and look into other alternatives, as the script relies on back-end queries it wouldn't need an update.

**Comment by Joey0x646576** [comment]
I've updated it to support any of the 'huggingface' conversation type models. 

However, currently one random model is selected as I cannot update the script to let you choose freely between e.g. Cleverbot and the HuggingFace alternatives. This is due the fact that I no longer use Hanbot. I can't test it in-game, but it should still function fine. Let me know how it goes :)

**Comment by Belzedar94** [comment]
It's working like a charm!!! I love the new model *-*

Out of curiosity, which model did you select from huggingface? It's 100 times better than Cleverbot. If I have to give negative feedback, is that sometimes it's kind of formal, but I love it!!

**Comment by Belzedar94** [comment]
By the way, I've been reading a bit and seems like the most advanced model in hugging face is:

https://huggingface.co/EleutherAI/gpt-neox-20b

**Comment by mrhammerthis** [comment]
yeah dude, would absolutely love this, was using another chat bot but it's been abandoned. Please update this as soon as you can. 

I appreciate you so much for making this (easy to get kills when people are replying lmao)
